hack/util.sh

```diff
@@ -8,116 +8,174 @@ TIME_MIN=$((60 * $TIME_SEC))
+	--hostname="${KUBELET_HOST}" \
+	--volume-dir="${VOLUME_DIR}" \
+	--etcd-dir="${ETCD_DATA_DIR}" \
+	--images="${USE_IMAGES}"
+
+
+	# Make oc use ${MASTER_CONFIG_DIR}/admin.kubeconfig, and ignore anything in the running user's $HOME dir
+	export ADMIN_KUBECONFIG="${MASTER_CONFIG_DIR}/admin.kubeconfig"
+	export CLUSTER_ADMIN_CONTEXT=$(oc config view --config=${ADMIN_KUBECONFIG} --flatten -o template -t '{{index . "current-context"}}')
+	sudo chmod -R a+rwX "${ADMIN_KUBECONFIG}"
+	echo "[INFO] To debug: export KUBECONFIG=$ADMIN_KUBECONFIG"
 }
 
 
 # start_os_server starts the OS server, exports the PID of the OS server
 # and waits until OS server endpoints are available
 function start_os_server {
-    echo "[INFO] Scan of OpenShift related processes already up via ps -ef  | grep openshift : "
-    ps -ef | grep openshift
-    echo "[INFO] Starting OpenShift server"
-    sudo env "PATH=${PATH}" OPENSHIFT_PROFILE=web OPENSHIFT_ON_PANIC=crash openshift start \
+	echo "[INFO] `openshift version`"
+	echo "[INFO] Server logs will be at:    ${LOG_DIR}/openshift.log"
+	echo "[INFO] Test artifacts will be in: ${ARTIFACT_DIR}"
+	echo "[INFO] Volumes dir is:            ${VOLUME_DIR}"
+	echo "[INFO] Config dir is:             ${SERVER_CONFIG_DIR}"
+	echo "[INFO] Using images:              ${USE_IMAGES}"
+	echo "[INFO] MasterIP is:               ${MASTER_ADDR}"
+
+	mkdir -p ${LOG_DIR}
+
+	echo "[INFO] Scan of OpenShift related processes already up via ps -ef	| grep openshift : "
+	ps -ef | grep openshift
+	echo "[INFO] Starting OpenShift server"
+	sudo env "PATH=${PATH}" OPENSHIFT_PROFILE=web OPENSHIFT_ON_PANIC=crash openshift start \
 	 --master-config=${MASTER_CONFIG_DIR}/master-config.yaml \
 	 --node-config=${NODE_CONFIG_DIR}/node-config.yaml \
 	 --loglevel=4 \
-    &> "${LOG_DIR}/openshift.log" &
-    export OS_PID=$!
-
-    echo "[INFO] OpenShift server start at: "
-    echo `date`
-  
-    wait_for_url "${KUBELET_SCHEME}://${KUBELET_HOST}:${KUBELET_PORT}/healthz" "[INFO] kubelet: " 0.5 60
-    wait_for_url "${API_SCHEME}://${API_HOST}:${API_PORT}/healthz" "apiserver: " 0.25 80
-    wait_for_url "${API_SCHEME}://${API_HOST}:${API_PORT}/healthz/ready" "apiserver(ready): " 0.25 80
-    wait_for_url "${API_SCHEME}://${API_HOST}:${API_PORT}/api/v1/nodes/${KUBELET_HOST}" "apiserver(nodes): " 0.25 80
-    
-    echo "[INFO] OpenShift server health checks done at: "
-    echo `date`
+	&> "${LOG_DIR}/openshift.log" &
+	export OS_PID=$!
+
+	echo "[INFO] OpenShift server start at: "
+	echo `date`
+	
+	wait_for_url "${KUBELET_SCHEME}://${KUBELET_HOST}:${KUBELET_PORT}/healthz" "[INFO] kubelet: " 0.5 60
+	wait_for_url "${API_SCHEME}://${API_HOST}:${API_PORT}/healthz" "apiserver: " 0.25 80
+	wait_for_url "${API_SCHEME}://${API_HOST}:${API_PORT}/healthz/ready" "apiserver(ready): " 0.25 80
+	wait_for_url "${API_SCHEME}://${API_HOST}:${API_PORT}/api/v1/nodes/${KUBELET_HOST}" "apiserver(nodes): " 0.25 80
```

Comment:
> These print a lot of garbage to stdout, would be nice to see it only on failure.

I think someone broke that recently.  I'll look up the pull to see why they did it.